servers/physics_3d/godot_shape_3d.cpp

```diff
@@ -1508,15 +1505,16 @@ _Volume_BVH *_volume_build_bvh(_Volume_BVH_Element *p_elements, int p_size, int
 	}
 
 	int split = p_size / 2;
-	bvh->left = _volume_build_bvh(p_elements, split, count);
-	bvh->right = _volume_build_bvh(&p_elements[split], p_size - split, count);
+	bvh.left = _volume_build_bvh(p_elements, split, count, pool);
+	bvh.right = _volume_build_bvh(&p_elements[split], p_size - split, count, pool);
 
-	//printf("branch at %p - %i: %i\n",bvh,count,bvh->face_index);
+	//printf("branch at %p - %i: %i\n",bvh_ptr,count,bvh.face_index);
 	count++;
-	return bvh;
+	*bvh_ptr = bvh;
+	return bvh_ptr;
 }
 
-void GodotConcavePolygonShape3D::_fill_bvh(_Volume_BVH *p_bvh_tree, BVH *p_bvh_array, int &p_idx) {
+void GodotConcavePolygonShape3D::_fill_bvh(_Volume_BVH *p_bvh_tree, BVH *p_bvh_array, int &p_idx, PagedAllocator<_Volume_BVH, false> &pool) {
```

Comment:
```suggestion
void GodotConcavePolygonShape3D::_fill_bvh(_Volume_BVH *p_bvh_tree, BVH *p_bvh_array, int &p_idx, PagedAllocator<_Volume_BVH, false> &r_pool) {
```

(`p_` prefix for parameters we only read, `r_` prefix for the ones we actually mutate)

Reply:
Interesting. When I see `&` without a `const` I always assume it's going to be mutated already^^